### DISCUSSION TITLE
Provide better structure for Views folder?

### DISCUSSION THREAD

**Discussion by d-miles** [OrchardCMS/OrchardCore#15834] [Q&A]
I'd like to better organize my Views folder. All of my main ContentTypes have names like:
* `Content-Service.liquid`
* `Content-HomePage.liquid`
* `Content-CarouselItem.liquid`

And their subparts are:
* `Service-BodyPart.liquid`
* `HomePage-Announcements.liquid` (named BagPart)

So I've got ~65 views total right now, most of them starting with "Content-", and the rest are associated with their parent, so named "Service-" for example. I'd like to be able to organize my views better. Something more MVC-like if possible:

* `{ContentType}/Content.liquid` -> equivalent of `Content-{ContentType}.liquid`
* `{ContentType}/ListPart.liquid` -> equivalent of `{ContentType}-ListPart.liquid`
* `{ContentType}/{SubContentType}.liquid` -> not sure of the equivalent, but if `{ContentType}` has a ListPart and `{SubContentType}` is a type within that list, I'd like to organize that type here as well as it is specific to the containing `{ContentType}`.

Any thoughts on the possibility of this? Basically taking the alternates file structure a bit further and creating paths instead, from what I can see. Could be optional.

**Comment by sebastienros** [comment]
Could you list all the files you created, and if possible group them by shape type, I want to understand how the hierarchy would look like in the end. For instance would we get 65 folder with one file in each with your suggestion?

Something we tested by the past is to move all shape templates in a `Shape` folder at least.

**Comment by d-miles** [comment]
I've not created any of my own shapes and instead just have created my own Content Types -- each area of the site is different and some have their own landing page and some don't. Here's a few examples of groups of views that I'd like to have grouped together, but can't be grouped together by default:

* Content-HomePage.liquid
* HomePage-Announcements.liquid
* HomePage-Gala.liquid
* Content-Service.HomePageSummary.liquid

---

* Content-LearnPage.liquid has a ListPart with the following items:
    * Content-LearnArticle.liquid
    * Content-LearnVideo.liquid
    * Content-LearnDocument.liquid
* Learn.Wrapper.liquid (wraps the ListPart items when viewing LearnPage)
* LearnPage-Definitions.liquid (BagPart on LearnPage)
* LearnPage-ListPart.liquid

---

* Content-NewsPage.liquid
* Content-NewsArticle.liquid
* Content-NewsArticle.Summary.liquid
* NewsArticle.Wrapper.liquid
* NewsPage-ListPart.liquid

---

* Content-ServicePage.liquid
* Content-Service.liquid
* Content-Service.Summary.liquid
* Service-BodyPart.liquid
* ServicePage.Wrapper.liquid
* ServicePage-ListPart.liquid

I'm thinking about just adding a new `IShapeTableProvider` which will look at the ContentType name and add some alternates myself so that I can organize things a bit better (basically removing `Content-` from a lot of things):

* ServicePage.liquid
* ServicePage.Wrapper.liquid
* ServicePage-Service.liquid (because it belongs to the ListPart of ServicePage)
* ServicePage-Service-BodyPart.liquid

I'm not sure that I have built my site using Orchard in the prescribed way, I basically just tried to figure out what would work and what woulddn't, and this is what I was able to come up with. I need stricter rules on page layouts and design to preserve the integrity of the site and provide consistency, so I'm not allowing users to create a `Page` or a `FlowPage` or anything like that on their own. I created a Content-like concept called Sections which each page has a BagPart of, and they include things like (short list because there are lots of these):

* Section.Wrapper.liquid (wraps all sections)
* ContainerSectionItem.Wrapper.liquid (wraps all sections within ContainerSection)
* Content-ContentSection.liquid
* Content-ContainerSection.liquid (has its own BagPart of a subset of Sections)
* Content-CollapsibleSection.liquid
* Content-CollapsibleSectionItem.liquid (has its own BagPart of nearly all sections Sections)
* Content-ImageSection.liquid
* Content-QuoteSection.liquid
* Content-TableSection.liquid
* Content-TableSectionRow.liquid
* Content-TitleSection.liquid
* _etc..._

I attempted to create a Shape called a Section and set the stereotype of each of these content types to this, but it proved too complicated because I'd have to provide editors for all Sections when I really just want the default Content editors, nothing special. So while there is a way for me to use Orchard to "fix" my view problem, I think the additional work is not worth the trouble. I was going to create a new Shape in Orchard for each of my pages but couldn't get all of the editors to work (even when just duplicating the `Content` ones from the source code. So I gave up on this.
